src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -376,19 +377,18 @@ class MigrationTo0_16(groupRepository: GroupRepository, appRepository: AppReposi
 class MigrationTo1_2(deploymentRepository: DeploymentRepository) {
   private[this] val log = LoggerFactory.getLogger(getClass)
 
-  def migrate(): Future[Unit] = {
+  def migrate(): Future[Unit] = async {
     log.info("Start 1.2 migration")
 
-    deploymentRepository.store.names().map(_.filter(deploymentRepository.isVersionKey)).flatMap { versionNodes =>
-      versionNodes.foldLeft(Future.successful(())) { (future, versionNode) =>
-        future.flatMap { _ =>
-          deploymentRepository.store.expunge(versionNode).map(_ => ())
-        }
-      }
-    }.flatMap { Unit =>
-      log.info("Finished 1.2 migration")
-      Future.successful(())
+    val nodes: Seq[String] = await(deploymentRepository.store.names())
+    val deploymentVersionNodes = nodes.filter(deploymentRepository.isVersionKey)
+
+    val it = deploymentVersionNodes.iterator
```

Comment:
Looking at `ZkStore` and `MarathonStore`, it seems that they use the default execution context.

I think we should merge this to see if it helps with the flaky unit test, even if it will be replaced by something different very soon.